src/Mono.Android/Android.Runtime/AndroidRuntime.cs

```diff
@@ -225,6 +227,66 @@ protected override IEnumerable<string> GetSimpleReferences (Type type)
 			return base.GetSimpleReferences (type)
 				.Concat (Enumerable.Repeat (j, 1));
 		}
+
+		delegate Delegate GetCallbackHandler ();
+
+		static MethodInfo dynamic_callback_gen;
+
+		static Delegate CreateDynamicCallback (MethodInfo method)
+		{
+			if (dynamic_callback_gen == null) {
+				var assembly = Assembly.Load ("Mono.Android.Export");
+				if (assembly == null)
+					throw new InvalidOperationException ("To use methods marked with ExportAttribute, Mono.Android.Export.dll needs to be referenced in the application");
+				var type = assembly.GetType ("Java.Interop.DynamicCallbackCodeGenerator");
+				if (type == null)
+					throw new InvalidOperationException ("The referenced Mono.Android.Export.dll does not match the expected version. The required type was not found.");
+				dynamic_callback_gen = type.GetMethod ("Create");
+				if (dynamic_callback_gen == null)
+					throw new InvalidOperationException ("The referenced Mono.Android.Export.dll does not match the expected version. The required method was not found.");
+			}
+			return (Delegate)dynamic_callback_gen.Invoke (null, new object [] { method });
+		}
+
+		public override void RegisterNativeMembers (JniType jniType, Type type, string methods)
+		{
+			try {
+				base.RegisterNativeMembers (jniType, type, methods);
+				return;
+			} catch (NotSupportedException e) {
+				// ignore, legacy fallback bellow
+			}
+
+			if (methods == null)
```

Comment:
This should be `string.IsNullOrEmpty(methods)`.

Reply:
The empty case would cause return as well in the next condition and testing against null saves a bit of code size here, in legacy part which will hopefully not be used much.

Reply:
Yes, empty case will return on the next line, as `members` will have no elements, but it'll require constructing a zero-length array in order to do that comparison. Using `string.IsNullOrEmpty()` will avoid the array creation if the string is empty.

That said, empty strings *shouldn't happen* in the legacy codepath, so this likely isn't necessary.